lib/promoted/ruby/client/validator.rb

```diff
@@ -145,7 +145,8 @@ def validate_fields!(obj, obj_name, fields)
                         raise ValidationError.new(field[:name].to_s + " is required on " + obj_name) if !obj.has_key?(field[:name])
                     end
 
-                    if field[:type] && obj.has_key?(field[:name]) then
+                    # If a field is provided as non-nil, it should be of the correct type.
+                    if field[:type] && obj.has_key?(field[:name]) && obj[field[:name]] != nil then
                         raise ValidationError.new(field[:name].to_s + " should be a " + field[:type].to_s) if !obj[field[:name]].is_a?(field[:type])
```

Comment:
side-comment - is there a benefit of also having the trailing if embedded on the raise line when there is already an if with conditions?

Reply:
Just style, the way I wrote it, the initial "if" is the preconditions, and the raise line "if" is the thing the business logic is actually checking for. I don't have a great preference.